frontend/packages/dev-console/src/components/pipelines/modals/triggers/submit-utils.ts

```diff
@@ -0,0 +1,91 @@
+import { RouteModel, ServiceModel } from '@console/internal/models';
+import { errorModal } from '@console/internal/components/modals';
+import { k8sCreate, k8sGet, K8sResourceKind, RouteKind } from '@console/internal/module/k8s';
+import { EventListenerModel, TriggerTemplateModel } from '../../../../models';
+import { Pipeline, PipelineRun } from '../../../../utils/pipeline-augment';
+import {
+  EventListenerKind,
+  TriggerTemplateKind,
+  TriggerTemplateKindParam,
+} from '../../resource-types';
+import { getPipelineRunFromForm } from '../common/utils';
+import {
+  createEventListener,
+  createEventListenerRoute,
+  createTriggerTemplate,
+} from './resource-utils';
+import { AddTriggerFormValues } from './types';
+
+const exposeRoute = async (elName: string, ns: string, iteration = 0) => {
+  const elResource: EventListenerKind = await k8sGet(EventListenerModel, elName, ns);
+  const serviceGeneratedName = elResource?.status?.configuration.generatedName;
+
+  if (!serviceGeneratedName) {
+    if (iteration < 3) {
+      setTimeout(() => exposeRoute(elName, ns, iteration + 1), 500);
+    } else {
+      // Unable to deterministically create the route; create a default one
+      await k8sCreate(RouteModel, createEventListenerRoute(elResource), { ns });
```

Comment:
Would be a good idea to wrap this up in try..catch as well.

Reply:
It's a fallback on a separate thread, but for good coding practices, I'll do that. Shouldn't whitescreen.